### PR TITLE
refactor: kernel files tree

### PR DESCRIPTION
**Pull request by alisterd51**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -35,7 +35,13 @@ ${addsuffix .rs, \
 	${addprefix src/, \
 		io \
 		lib \
-		vga_buffer \
+		${addprefix driver/, \
+			mod \
+			${addprefix vga/, \
+				mod \
+				text_mode \
```
Comment (JonathanDUFOUR) [inline Makefile:42]: Why `mode`?

File: src/driver/vga/mod.rs [hunk 2]
```diff
@@ -0,0 +1 @@
+pub mod text_mode;
```
Comment (JonathanDUFOUR) [inline src/driver/vga/mod.rs:1]: Why `mode`?

**Review by JonathanDUFOUR — COMMENTED** [review]
Please answer my comments before I approve it.

File: Makefile (hunk 1 above)
Comment (alisterd51) [reply Makefile:42]: text_mode refers to the [doc](https://wiki.osdev.org/Text_mode)

Comment (JonathanDUFOUR) [reply Makefile:42]: Why not name it `text_ui` or `text_user_interface` then?

**Pull request merged**